python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -1064,7 +1064,7 @@ def transfer_from(
             # ask the receiving datastore to copy it when it doesn't exist
             # so we have to filter again based on what the source datastore
             # understands.
-            known_to_source = source_child.knows_these([ref for ref in refs])
+            known_to_source = source_child.knows_these(list(refs))
```

Comment:
Not sure you need `list` here, `knows_these` says it takes `Iterable`, and `refs` is Iterable here too. I think this code already assumes that you can iterate many times over `refs`, which is probably not true for `Iterable` in general.

Reply:
I think I was forcing it to a list originally so that we could go through it multiple times, which probably means the annotation is wrong. Should it be Sequence?

Reply:
We can keep `Iterable` but then make sure that code can handle single-pass iterables. This code in chainedDatastore cannot, it calls `set(refs)`  few lines above. fileDatastore also douig similar thing in `knows_these`. Or alternatively we should use Sequence everywhere.